src/DurableTask.Netherite/StorageLayer/Faster/AzureBlobs/AzureStorageDevice.cs

```diff
@@ -555,45 +558,62 @@ async Task ReadFromBlobAsync(UnmanagedMemoryStream stream, BlobUtilsV12.PageBlob
                     offset += length;
                 }
             }
+            catch (Exception e)
+            {
+                IsFaulted = true;
+
+                // details about failed storage accesses are already traced by the error handler, but we still create another trace here
+                // to help us debug situations where exceptions are thrown in other places, and to clarify the execution path
+                this.BlobManager.StorageTracer?.FasterStorageError($"{nameof(ReadFromBlobAsync)} id={id} encountered exception", e);
+            }
+            finally
+            {
+                stream.Dispose();
+            }
```

Comment:
Is there a reason to explicitly dispose the `stream` instead of continue to use the `using(stream)` statement, as before?

Reply:
Yeah, I did not like to have both `try` and `using` - too much nesting.